crates/weaver_resolver/src/registry.rs

```diff
@@ -115,9 +133,97 @@ pub fn resolve_semconv_registry(
         })
         .collect();
 
+    // Check the `any_of` constraints.
+    let attr_name_index = attr_catalog.attribute_name_index();
+    check_any_of_constraints(&ureg.registry, &attr_name_index)?;
+
+    // All constraints are satisfied.
+    // Remove the constraints from the resolved registry.
+    for group in ureg.registry.groups.iter_mut() {
+        group.constraints.clear();
+    }
+
     Ok(ureg.registry)
 }
 
+/// Checks the `any_of` constraints in the given registry.
+///
+/// # Arguments
+///
+/// * `registry` - The registry to check.
+/// * `attr_name_index` - The index of attribute names (catalog).
+///
+/// # Returns
+///
+/// This function returns `Ok(())` if all the `any_of` constraints are satisfied.
+/// Otherwise, it returns the error `Error::UnsatisfiedAnyOfConstraint`.
+pub fn check_any_of_constraints(
+    registry: &Registry,
+    attr_name_index: &[String],
+) -> Result<(), Error> {
+    for group in registry.groups.iter() {
+        // Build a list of attribute names for the group.
+        let mut group_attr_names = HashSet::new();
+        for attr_ref in group.attributes.iter() {
+            let attr_name =
+                attr_name_index
+                    .get(attr_ref.0 as usize)
+                    .ok_or(Error::UnresolvedAttribute {
+                        attribute_ref: *attr_ref,
+                    })?;
+            group_attr_names.insert(attr_name.clone());
+        }
+
+        check_group_any_of_constraints(
+            group.id.as_ref(),
+            group_attr_names,
+            group.constraints.as_ref(),
+        )?;
+    }
+
+    Ok(())
+}
+
+/// Checks the `any_of` constraints for the given group.
+fn check_group_any_of_constraints(
+    group_id: &str,
+    group_attr_names: HashSet<String>,
+    constraints: &[Constraint],
+) -> Result<(), Error> {
+    let mut any_of_unsatisfied = 0;
+    let mut any_of_total = 0;
+    let mut any_of_constraints = vec![];
+    'outer: for constraint in constraints.iter() {
+        if constraint.any_of.is_empty() {
+            continue;
+        }
+        any_of_total += 1;
+
+        // Check if the group satisfies the `any_of` constraint.
+        any_of_constraints.push(constraint.any_of.clone());
+        for attr_name in constraint.any_of.iter() {
```

Comment:
I think you can get rid of `'outer` if you do:

```
if constraint.any_of.iter().find(|name| !group_attr_names.contains(name)).is_some() {
  any_of_unsatisified += 1
}
```

If you need to know which isn't found, you could make a new error type to aggregate:

```
if let Some(attr) = constraint.any_of.iter().find(|name| !group_attr_names.contains(name)) {
  any_of_unsatisfied += 1;
  missing_attribute_errors.push(MissingAnyOfError::new(attr, constraint));
}
```

Reply:
Simplified the code to better handle this type of error